Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/DataEditorForm.java

```diff
@@ -190,9 +193,10 @@ public String open(int id, String referringView) {
             this.currentChildren.addAll(process.getChildren());
             this.user = ServiceManager.getUserService().getCurrentUser();
 
-            ruleset = openRulesetFile(process.getRuleset().getFile());
+            ruleset = openRuleset(process.getRuleset());
             openMetsFile();
             init();
+            openProcesses.put(Integer.valueOf(id), user);
```

Comment:
Are you sure? If I write the process number to the map with a primitive `int` and request lookup with an Integer _object_, couldn’t it _not_ find it because it's the wrong class? I'm not sure.

Reply:
1. int is not a class, it is primitive
2. there exists something called autoboxing (https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html)

_Consider the following code:_

```
List<Integer> li = new ArrayList<>();
for (int i = 1; i < 50; i += 2)
    li.add(i);
```
_Although you add the int values as primitive types, rather than Integer objects, to li, the code compiles. Because li is a list of Integer objects, not a list of int values, you may wonder why the Java compiler does not issue a compile-time error. The compiler does not generate an error because it creates an Integer object from i and adds the object to li. Thus, the compiler converts the previous code to the following at runtime:_

```
List<Integer> li = new ArrayList<>();
for (int i = 1; i < 50; i += 2)
    li.add(Integer.valueOf(i));
```

3. process.getId() returns Integer, not int

### Summary
Considering above points, you don't need call Integer.valueOf(int).

Reply:
fixed